### ISSUE TITLE
C++: SIGABRT instead of IndexError for out-of-bounds

### ISSUE THREAD

**Issue by mratsim** [nim-lang/Nim#6512]
C++ backend bounds checking does not work for seq

```Nim
let a = newSeq[int](3)

discard a[3]
```

Compiling with nim c
```Nim
Hint: ./bin/seq_bound_check  [Exec]
Traceback (most recent call last)
seq_bound_check.nim(5)   seq_bound_check
system.nim(2739)         sysFatal
Error: unhandled exception: index out of bounds [IndexError]
```

Compiling with nim cpp
```Nim
Hint: ./bin/seq_bound_check  [Exec]
libc++abi.dylib: terminating with uncaught exception of type NimException
Traceback (most recent call last)
seq_bound_check.nim(5)   seq_bound_check
system.nim(2739)         sysFatal
SIGABRT: Abnormal termination.
```

**Comment by Yardanico** [comment]
@mratsim it's not related to bounds checking, it's actually related to ALL exceptions when in C++ mode

**Comment by Araq** [comment]
Works with 0.18.

**Comment by mratsim** [comment]
Fixed in 0.18
